src/editor/contentState/selectionCtrl.js

```diff
@@ -0,0 +1,58 @@
+const selectionCtrl = ContentState => {
+  // Returns the table from the table cell:
+  //   table <-- thead or tbody <-- tr <-- th or td (cell)
+  ContentState.prototype.getTableFromTableCell = function (block) {
+    const table = this.getParent(this.getParent(this.getParent(block)))
+    if (table && table.type !== 'table') {
+      return null
+    }
+    return table
+  }
+
+  ContentState.prototype.tableCellHandler = function (event) {
+    const { start, end } = this.cursor
+    const startBlock = this.getBlock(start.key)
+    const endBlock = this.getBlock(end.key)
+    const { type: startType } = startBlock
+    const { type: endType } = endBlock
+    if (/th|td/.test(startType) && /th|td/.test(endType)) {
+      if (start.key === end.key) {
+        const { text, key } = startBlock
+        this.cursor = {
+          start: { key, offset: 0 },
+          end: { key, offset: text.length }
+        }
+      } else {
+        const startTable = this.getTableFromTableCell(startBlock)
+        const endTable = this.getTableFromTableCell(endBlock)
+        // Check whether both blocks are in the same table.
+        if (!startTable || !endTable) {
+          console.error('No table found or invalid type.')
+          return
+        } else if (startTable.key !== endTable.key) {
+          // Select entire document
+          return
+        }
+        const firstTableBlock = this.firstInDescendant(startTable)
+        const lastTableBlock = this.lastInDescendant(startTable)
+        if (!firstTableBlock || !/th|td/.test(firstTableBlock.type) ||
+          !lastTableBlock || !/th|td/.test(lastTableBlock.type)) {
+          console.error('No table cell found or invalid type.')
+          return
+        }
+        const { key: startKey } = firstTableBlock
```

Comment:
Set the `cursor.start` in the first cell of the table, will solve the `text disappear` bug?

Reply:
@fxha 
>Sometimes when I select two table cells and press Ctrl+A the selected text disappears. I think it's because of this.cursor but I don't know.

What's the exactly meaning of `text disappears`? and how to reproduce it?

Reply:
OK, another write-timeout bug which was solved with a rebase... :cry: I think we can merge it.